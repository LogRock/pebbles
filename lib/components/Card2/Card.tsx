import styled from "styled-components";
import { Shadows, Spacings } from "../../types/theme";

export interface CardProps {
  shadow?: keyof Shadows;
  padding?: keyof Spacings;
  radius?: keyof Spacings;
}

const Card2 = styled.div<CardProps>`
  padding: ${({ theme, padding }) => theme.spacings[padding || "md"]};

  border-radius: ${({ theme, radius }) => theme.spacings[radius || "sm"]};

  background: ${({ theme }) => theme.colors.shades[0]};
  box-shadow: ${({ theme, shadow }) => theme.shadows[shadow || "small"]};
`;

export default Card2;
