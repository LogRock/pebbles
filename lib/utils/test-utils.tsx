/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
