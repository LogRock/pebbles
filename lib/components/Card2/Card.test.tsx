import React from "react";
import { render } from "../../utils/test-utils";
import Card2 from "./Card";

describe("Card", () => {
  it("renders with default props", () => {
    const { container } = render(<Card2 />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom props", () => {
    const { container } = render(
      <Card2 padding="huge" shadow="upXxLarge" radius="xxhuge" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
