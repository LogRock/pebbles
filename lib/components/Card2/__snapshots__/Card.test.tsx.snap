// Vitest Snapshot v1

exports[`Card > renders with custom props 1`] = `
<div
  class="sc-bdvvtL fKFwXa"
  radius="xxhuge"
/>
`;

exports[`Card > renders with default props 1`] = `
<div
  class="sc-bdvvtL gCAtDt"
/>
`;
